build: New testsuite failures with 3.1.16.0

Package and build output can be found here

https://build.opensuse.org/package/show/home:darix:branches:graphics/OpenImageIO

We are still building with OpenCV 4.x - which could explain that problem, I guess.
But I am a bit puzzled about the OpenEXR problem.


3.1.15.0 package can be found here for comparison https://build.opensuse.org/package/show/graphics/OpenImageIO

```
head -n 10000 ./build/testsuite/openexr-partialtile/build.txt ./build/testsuite/imagebufalgo-opencv/build.txt
==> ./build/testsuite/openexr-partialtile/build.txt <==
-- The CXX compiler identification is GNU 16.1.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:17 (find_package):
  Could not find a package configuration file provided by "OpenImageIO" with
  any of the following names:

    OpenImageIO.cps
    openimageio.cps
    OpenImageIOConfig.cmake
    openimageio-config.cmake

  Add the installation prefix of "OpenImageIO" to CMAKE_PREFIX_PATH or set
  "OpenImageIO_DIR" to a directory containing one of the above files.  If
  "OpenImageIO" provides a separate development package or SDK, be sure it
  has been installed.


-- Configuring incomplete, errors occurred!
gmake: Makefile: No such file or directory
gmake: *** No rule to make target 'Makefile'.  Stop.

==> ./build/testsuite/imagebufalgo-opencv/build.txt <==
-- The CXX compiler identification is GNU 16.1.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:17 (find_package):
  Could not find a package configuration file provided by "OpenImageIO" with
  any of the following names:

    OpenImageIO.cps
    openimageio.cps
    OpenImageIOConfig.cmake
    openimageio-config.cmake

  Add the installation prefix of "OpenImageIO" to CMAKE_PREFIX_PATH or set
  "OpenImageIO_DIR" to a directory containing one of the above files.  If
  "OpenImageIO" provides a separate development package or SDK, be sure it
  has been installed.


-- Configuring incomplete, errors occurred!
gmake: Makefile: No such file or directory
gmake: *** No rule to make target 'Makefile'.  Stop.
```

This is probably another case like this test, which we already skip

```
# cmake-consumer docs-examples-cpp -> currently failing tests as they assume normal cmake search paths will work to find the OIIO devel files in the final location
```

This was the outcome of our last super long testsuite debug round.

I will also skip those 2 tests just like cmake-consumer for now?

--- a/src/cmake/testing.cmake
+++ b/src/cmake/testing.cmake
@@ -103,9 +103,15 @@ macro (oiio_add_tests)
             set (_test_disabled TRUE)
         endif ()
     endforeach ()
-    # For OCIO 2.2+, have the testsuite use the default built-in config
+    # Things we add to the environment for tests:
+    # - For OCIO 2.2+, have the testsuite use the default built-in config.
+    # - Some tests (e.g. cmake-consumer) configure their own child cmake
+    #   project that does find_package(OpenImageIO). Point them at this
+    #   build's install location so that works out of the box, without users
+    #   needing to set OpenImageIO_ROOT in their environment themselves.
     list (APPEND _ats_ENVIRONMENT "OCIO=ocio://default"
-                                  "OIIO_TESTSUITE_OCIOCONFIG=ocio://default")
+                                  "OIIO_TESTSUITE_OCIOCONFIG=ocio://default"
+                                  "OpenImageIO_ROOT=${CMAKE_INSTALL_PREFIX}")
     if (_test_disabled)
         message (STATUS "Skipping test(s) ${_ats_UNPARSED_ARGUMENTS} because of disabled ${_ats_ENABLEVAR}")
     elseif (_test_notfound)

--- a/testsuite/runtest.py
+++ b/testsuite/runtest.py
@@ -426,6 +426,11 @@ def runtest (command: str, outputs: list[str], failureok: int=0) -> int :
             print ("#### Error: this command failed: ", sub_command)
             print ("FAIL")
             err = 1
+            if os.path.isfile("build.txt") :
+                print ("---   BUILD LOG   ---\n")
+                with open("build.txt", "r") as fbuild :
+                    print (fbuild.read())
+                print ("--- END BUILD LOG ---\n")
 
     for out in outputs :
         (prefix, extension) = os.path.splitext(out)

